### PR TITLE
Version for webpack 4

### PR DESCRIPTION
**Pull request by stackdumper**
## Purpose

Webpack 4 introduces [breaking changes](https://github.com/webpack/webpack/releases/tag/v4.0.0) for plugins/loaders.

## Additional

- Add example app (`example/`)

### Related issues

#33

### CODE REVIEW

File: src/types.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+export type PrepackConfigType = Object;
 
-export type PluginConfigurationType = {
-  prepack: PrepackConfigurationType,
+export type PluginConfigType = {
+  prepack: PrepackConfigType,
   test: RegExp
 };
 
-export type UserPluginConfigurationType = {
```
Comment (gajus) [inline src/types.js]: Please don't start changing variable names in project based on personal preference.

The convention is to use full names, regardless of how long.

**Review by gajus — CHANGES_REQUESTED** [review]

File: src/types.js (hunk 1 above)
Comment (stackdumper) [reply src/types.js]: Fixed in a61d42b

**Review by gajus — APPROVED** [review]

**Comment by gajus** [comment]
Thank you.

I have added a quick README as well.

https://github.com/gajus/prepack-webpack-plugin/tree/master/example

**Comment by stackdumper** [comment]
@gajus note that this version is compatible only with **webpack >= 4**.

I think we should provide different package versions for different webpack versions. For example, `v1` for `webpack <= 3` and `v2` for `webpack >= 4`.

**Comment by gajus** [comment]
Considering how bleeding edge prepack is most people who are going to be using prepack are also going to be using webpack 4.

Maintaining two versions with semantic-release is not easy/ possible.

**Comment by denji** [comment]
prepack-webpack-plugin for webpack@4 has not yet been published to the npm registry?

> "prepack-webpack-plugin": "^1.1.0",

**Comment by MattyBalaam** [comment]
Is there any help anyone can do to get the 2.0 version published?

**Comment by gajus** [comment]
> Is there any help anyone can do to get the 2.0 version published?

There is no version 2. It has been released as v1.1.0 version.

**Comment by hipstersmoothie** [comment]
I'm still seeing https://github.com/gajus/prepack-webpack-plugin/issues/33 even with v1.1.0

**Comment by hipstersmoothie** [comment]
@gajus just checked the PrepackPlugin.js in my node modules and it doesn't show the changes made in this PR.

**Comment by stackdumper** [comment]
As I know, there is no possibility to publish a package to NPM without version update. The current version of NPM package is `1.1.0` (was released a year ago).

I think updating version to `1.2.0` should solve release problem.

**Comment by hipstersmoothie** [comment]
yeah otherwise I think you would have to unpublish and republish.

Might as well just increment the version. Even a patch would be fine because it isn't a minor change. Since 1.1.0 was an unsuccessful publish, this could be 1.1.1

**Pull request merged**